pep-0724.rst

```diff
@@ -0,0 +1,235 @@
+
+Since ``list`` is invariant, it doesn't have any subtypes. ``list[str]`` is not
+a subtype of ``list[str | int]``. This means type checkers cannot narrow the
+type to ``list[int]`` in the false case.
+
+Type checkers should not assume any narrowing in the false case when the
+``TypeGuard`` type argument is not a subtype of the first input parameter type. 
+
+However, narrowing in the true case is still possible. In the example above, the
+type checker can assume the list is a ``list[str]`` if the ``TypeGuard``
+function returns true.
+
+User error
+--------------------------
+
+The new ``else`` case for a ``TypeGuard`` can be setup incorrectly. Here's an
+example:
+
+.. code-block:: python
+
+    def is_positive_int(val: int | str) -> TypeGuard[int]:
+        return isinstance(val, int) and val > 0
+
+    def func(val: int | str):
+        if is_positive_int(val):
+            # Type checker assumes int here
+        else:
+            # Type checker assumes str here
+
+A type checker will assume for the else case that the value is ``str``. This is
+a change in behavior from :pep:`647` but as that pep stated, there are many ways
+a determined or uninformed developer can subvert type safety.
+
+A better way to handle this example would be something like so:
+
+.. code-block:: python
+
+    PosInt = NewType('PosInt', int)
+
+    def is_positive_int(val: PosInt | int | str) -> TypeGuard[PosInt]:
+        return isinstance(val, int) and val > 0
+
+    def func(val: int | str):
+        if is_positive_int(val):
+            # Type checker assumes PosInt here
+        else:
+            # Type checker assumes str | int here
+
+
+Backwards Compatibility
+=======================
+
+For preexisting code this PEP should require no changes.
+
+However, some use cases such as the one below can be simplified:
+
+.. code-block:: python
+
+    class A():
+        pass
+    class B():
+        pass
+
+    def is_A(x: A | B) -> TypeGuard[A]:
+        return is_instance(x, A)
+
+
+    def is_B(x: A | B) -> TypeGuard[B]:
+        return is_instance(x, B)
+
+
+    def test(x: A | B):
+        if is_A(x):
+            # Do stuff assuming x is an 'A'
+            return
+        assert is_B(x)
+
+        # Do stuff assuming x is a 'B'
+        return
+
+
+With this proposed change, the code above continues to work but could be
+simplified by removing the assertion that x is of type B in the negative case:
+
+.. code-block:: python
+
+    class A():
+        pass
+    class B():
+        pass
+
+    def is_A(x: A | B) -> TypeGuard[A]:
+        return is_instance(x, A)
+
+
+    def test(x: A | B):
+        if is_A(x):
+            # Do stuff assuming x is an 'A'
+            return
+
+        # Do stuff assuming x is a 'B'
+        return
+
+
+How to Teach This
+=================
+
+The belief is that new users will assume this is how ``TypeGuard`` works in the
+first place. Meaning this change should make ``TypeGuard`` easier to teach.
```

Comment:
A suggested re-wording which makes it clearer that this is an assertion / belief on the part of the authors, and that I think clarifies the intent here. 

```suggestion
We assert that users unfamiliar with ``TypeGuard`` will expect
the behavior outlined in this PEP,
therefore making ``TypeGuard`` easier to teach and explain.
```